apps/mobile/src/common/transactions/stacks-transactions.hooks.ts

```diff
@@ -0,0 +1,53 @@
+import { AccountId } from '@/models/domain.model';
+import { useAccountByIndex } from '@/store/accounts/accounts.read';
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+import { useNetworkPreferenceStacksNetwork } from '@/store/settings/settings.read';
+import { bytesToHex } from '@noble/hashes/utils';
+import { AnchorMode } from '@stacks/transactions';
+
+import { useNextNonce } from '@leather.io/query';
+import {
+  StacksUnsignedTokenTransferOptions,
+  TransactionTypes,
+  generateUnsignedTransaction,
+} from '@leather.io/stacks';
+import { createMoney } from '@leather.io/utils';
+
+export function useStxTokenTransferDetails({ fingerprint, accountIndex }: AccountId) {
+  const account = useAccountByIndex(fingerprint, accountIndex);
+  const network = useNetworkPreferenceStacksNetwork();
+  const stxSigner = useStacksSigners().fromAccountIndex(fingerprint, accountIndex)[0];
+  const stxAddress = stxSigner?.address ?? '';
+  const { data: nextNonce } = useNextNonce(stxAddress);
+
+  if (!account || !stxSigner) return;
```

Comment:
Any way to handle these missing values differently, such that by the time we get to this hook, we're sure they're defined?

Reply:
I ended up creating a loader for the StacksSigner and passing the signer address and publicKey hex thru the route params. Hoping this is a good solution here?